pandas/core/generic.py

```diff
@@ -2012,6 +2012,10 @@ def sample(self, n=None, frac=None, replace=False, weights=None, random_state=No
         # Check weights for compliance
         if weights is not None:
 
+            # If a series, align with frame
+            if isinstance(weights, pd.Series):
```

Comment:
you need tests for this. Further this might generate NaN's in the weights (which I think is ok).